sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/CustomProperties.java

```diff
@@ -0,0 +1,133 @@
+package com.microsoft.azure.mobile;
+
+import com.microsoft.azure.mobile.utils.MobileCenterLog;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Custom properties builder.
+ * Collect multiple properties for send its at once in the same log.
+ */
+public class CustomProperties {
+
+    private static final Pattern KEY_PATTERN = Pattern.compile("^[a-zA-Z][a-zA-Z0-9]*$");
+
+    /**
+     * Properties key/value pairs.
+     * Null value means that key marked to clear.
+     */
+    private Map<String, Object> properties = new HashMap<>();
```

Comment:
Everywhere except in JSON files (we'll change json later) we use the android field naming convention: mProperties, and not this. when accessing it.